src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+        deleteCarDriversRelations(car.getId());
+        addCarDriversRelations(car.getDrivers(), car.getId());
+        return car;
+    }
+
+    private void addCarDriversRelations(List<Driver> drivers, Long id) {
+        String query = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            for (Driver d : drivers) {
+                statement.setLong(1, d.getId());
+                statement.setLong(2, id);
+                statement.executeUpdate();
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't add new relation "
+                    + drivers + " in cars_drivers DB.", e);
+        }
+    }
+
+    private boolean deleteCarDriversRelations(Long id) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete relations for car with id " + id, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllCarsByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
+        List<Long> carsId = new ArrayList<>();
+        String getAllCarsByDriverQuery = "SELECT * FROM cars_drivers WHERE driver_id = ?";
```

Comment:
You should have SELECT with JOINs here to get all info and avoid multiple calling `get(ci)` 
And then just add drivers to each car.